container_api/src/config.rs

```diff
@@ -177,13 +177,14 @@ pub struct LoggerConfiguration {
 ///
 /// Projected are various DB adapters.
 #[derive(Deserialize, Clone)]
-pub struct StorageConfiguration {
-    #[serde(rename = "type")]
-    pub storage_type: String,
-    pub username: Option<String>,
-    pub password: Option<String>,
-    pub url: Option<String>,
-    pub path: Option<String>,
+#[serde(tag="type")]
+pub enum StorageConfiguration {
+    #[serde(rename = "memory")]
+    Memory,
+    #[serde(rename = "file")]
+    File {
+        path: String,
```

Comment:
Please make this a `PathBuf`

Reply:
@sphinxc0re I started doing this but if you follow the trail, this gets fed into several other functions including `create_context` which also take a `String` and should probably take a `Path`. I'd like to defer it to a separate small PR.

Reply:
😱 When a path is needed `PathBuf` should be used. This sounds like a followup.